dags/destinations/gads_oci.py

```diff
@@ -0,0 +1,353 @@
+"""
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License."""
+
+"""Google Ads OCI destination implementation."""
+import errors
+
+from collections import defaultdict
+from google.ads.googleads.errors import GoogleAdsException
+from pydantic import Field
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Tuple
+from utils import GoogleAdsUtils, ProtocolSchema, RunResult, ValidationResult
+
+_BATCH_SIZE = 10000
+
+_DEFAULT_CURRENCY_CODE = "USD"
+
+_REQUIRED_FIELDS = [
+  "customer_id",
+  "conversion_action_id",
+  "conversion_date_time",
+  "conversion_value",
+]
+
+_ID_FIELDS = [
+  "gclid",
+  "gbraid",
+  "wbraid",
+]
+
+ConversionIndicesToConversions = List[Tuple[int, Any]]
+CustomerConversionMap = Dict[str, ConversionIndicesToConversions]
+InvalidConversionIndices = List[Tuple[int, errors.ErrorNameIDMap]]
+PartialFailures = Dict[int, str]
+
+
+class Destination:
+  """Implements DestinationProto protocol for Google Ads OCI."""
+
+  def __init__(self, config: Dict[str, Any]) -> None:
```

Comment:
Remove "-> None" to match the [DestinationProto](https://github.com/google-marketing-solutions/Tightlock/blob/main/dags/protocols/destination_proto.py) signature. I think this is the cause for the failing test.